src/DynamoCore/Graph/Nodes/ZeroTouch/DSFunctionBase.cs

```diff
@@ -40,11 +40,15 @@ public override bool IsConvertible
 
         public override IdentifierNode GetAstIdentifierForOutputIndex(int outputIndex)
         {
-            return Controller.ReturnKeys != null && Controller.ReturnKeys.Any()
-                ? base.GetAstIdentifierForOutputIndex(outputIndex)
-                : (OutPortData.Count == 1
-                    ? AstIdentifierForPreview
-                    : base.GetAstIdentifierForOutputIndex(outputIndex));
+            if (Controller.ReturnKeys != null && Controller.ReturnKeys.Any())
+            {
+                string id = AstIdentifierBase + "_out" + outputIndex;
+                return AstFactory.BuildIdentifier(id);
+            }
+            else
+            {
+                return base.GetAstIdentifierForOutputIndex(outputIndex); 
+            }
```

Comment:
Just want to make sure this is the intended change (there wasn't a comment to explain to me), because it seems like the logic is reversed around the first condition in the original implementation (which is like the following):

```csharp
if (roller.ReturnKeys != null && Controller.ReturnKeys.Any() || 
    OutPortData.Count != 1)
{
    return base.GetAstIdentifierForOutputIndex(outputIndex);
}
else
{
    return AstIdentifierForPreview;
}
```